calculator/calculator.py

```diff
@@ -0,0 +1,218 @@
+"""Calculator module.
+
+This module allows evaluation of strings which represent mathematical
+expressions. The supported operations are declared in operators dictionary.
+
+Example:
+    To use this module, import the evaluate() function:
+
+        from calculator import evaluate
+
+    The parameter of this function is a mathematical expression (str). For
+    example, "1 + (3^-2) * (4! - 5$ 2)".
+"""
+
+import re
+import math
+import operator
+from sys import maxsize
+
+any_number_regex = r"-?\d+(\.\d+)?"
+
+
+class Operator:
+    """Representation of a single mathematical operator.
+
+    Attributes:
+        operation (function): the function that does the operation.
+        precedence (int): the priority of the operation.
+        regex (str): regex to find a pattern which suits to the operator.
+
+    Note:
+        Precedence determines which operation goes after another.
+    """
+
+    def __init__(self, operation, precedence, regex):
+        self.operation = operation
+        self.precedence = precedence
+        self.regex = re.compile(regex)
+
+
+def average(num1, num2):
+    """Return the average of two numbers (int/float).
+
+    Note:
+        Same as using mean() from statistics module.
+    """
+    return float(num1 + num2) / 2.0
+
+
+def remove_spaces(expression: str) -> str:
+    """Remove the spaces between characters in a string."""
+    return ''.join(expression.split())
+
+
+def get_sub_expression_by_symbol(symbol: str, expression: str) -> str:
+    """Find the sub expression the current operator.
+
+    Args:
+        symbol: the current operator symbol.
+        expression: the entire expression to search.
+
+    Returns:
+         The desired sub expression of the operator. If nothing is found
+         returns an empty string.
+    """
+    match = operators[symbol].regex.search(expression)
```

Comment:
in non-code english this line would be:
find if the operator.regex serach for an equation

but if you would verably explain youd say smth like:
search if there is an equation inculding the operator in the expression